### DISCUSSION TITLE
Upgrade CKAN on OpenShift

### DISCUSSION THREAD

**Discussion by mauricio-pmf** [ckan/ckan#7887] [Q&A]
Hello, everyone.

We have a multi-container environment (CKAN itself, DataPusher, Redis, Solr) running on OpenShift. It was deployed and managed by an independent company and only recently handed over to us. Unfortunately, no detailed documentation was provided as part of the implementation/customization. There is a customized theme in place but no extensions.

We suspect the deploymed was based on this repository:  https://github.com/dagniel/ckan-on-openshift/, which refers to version 2.8.2. We need to upgrade our environment to the latest version, but the official documentation does not provide detailed steps on how this must be done considering a multi-container scenario on OpenShift.

Can anyone guide us on the right way to succeed with the upgrade?

Thanks a lot.

**Comment by kowh-ai** [comment]
Maybe someone with good OpenShift experience can respond to your discussion question here. In the meantime are you able to ask specific questions on the **non-OpenShift** part of the upgrade requirements?

**Reply by mauricio-pmf** [reply]
Hey, Brett.

Thanks for replying to my post. Well, this is the link that talks about upgrading a minor release: https://docs.ckan.org/en/2.10/maintaining/upgrading/upgrade-package-to-minor-release.html. It mentions the steps to execute a migration on an apt/deb distro, and that considers CKAN is running as a monolithic application, which is not our case, as previously explained.

The upgrade from a source package (https://docs.ckan.org/en/2.10/maintaining/upgrading/upgrade-source.html) link does not take into account a multi-container scenario either.

Hence, I don't know how can I progress or ask further questions since my environment runs in multiple containers on top of OpenShift. :-)

Mauricio

**Reply by mauricio-pmf** [reply]
Ignoring the nuances of OpenShift, how an upgrade on a multiple container scenario (pure Docker, pure Podman, Kubernetes, AWS EKS, Azure AKS, ...) works?

**Reply by kowh-ai** [reply]
I think the best practice would be to build the docker image for CKAN independently of any orchestration platform. This is what we do for the official CKAN docker repo's. So when you need to upgrade to a new version of CKAN you would just change the image tag to the new (pre-built) image that exists in your docker registry (DockerHub for instance).
